app/models/attendance.rb

```diff
@@ -32,23 +34,94 @@ class Attendance < UuidApplicationRecord
     where('check_in BETWEEN ? AND ?', week.at_beginning_of_week(:sunday), week.at_end_of_week(:saturday))
   }
 
+  scope :absences, -> { where.not(absence: nil) }
+  scope :non_absences, -> { where(absence: nil) }
+
   scope :illinois_part_days, -> { where('total_time_in_care < ?', '5 hours') }
   scope :illinois_full_days, -> { where('total_time_in_care BETWEEN ? AND ?', '5 hours', '12 hours') }
   scope :illinois_full_plus_part_days, -> { where('total_time_in_care > ? AND total_time_in_care < ?', '12 hours', '17 hours') }
   scope :illinois_full_plus_full_days, -> { where('total_time_in_care BETWEEN ? AND ?', '17 hours', '24 hours') }
 
+  delegate :business, to: :child_approval
+  delegate :user, to: :child_approval
+  delegate :child, to: :child_approval
+  delegate :state, to: :child
+  delegate :county, to: :child
+  delegate :timezone, to: :user
+
   private
 
+  def round_check_in
+    return unless check_in
+
+    self.check_in = Time.at(check_in - check_in.sec).in_time_zone(timezone)
+  end
+
+  def round_check_out
+    return unless check_out
+
+    self.check_out = Time.at(check_out - check_out.sec).in_time_zone(timezone)
+  end
+
   def calc_total_time_in_care
     self.total_time_in_care = if check_in && check_out
-                                check_out - check_in
-                              elsif child_approval.child.state == 'NE'
+                                (check_out - check_in).round(2)
+                              elsif state == 'NE'
                                 calculate_from_schedule
                               else
                                 0.seconds
                               end
   end
 
+  def calc_earned_revenue
+    return unless state == 'NE'
+
+    self.earned_revenue = child_approval.special_needs_rate ? ne_special_needs_revenue : ne_base_revenue
+  end
+
+  def ne_hours
```

Comment:
We're already using the Hours and Days calculators in the dashboard, we're just repurposing the rounding to quarter hours and calculating full days based on duration methods to calculate revenue when an attendance is entered.